### ISSUE TITLE
Update internal documentation

### ISSUE THREAD

**Issue by rennergade** [Lind-Project/lind-wasm#43]
Now that we've made significant head way lets make an attempt to update our internal documentation.

Ruchir has offered to edit/package everything nicely for us so we should take him up on that, the rest of us need to provide him content.

I think the best breakdown is the checklist below, can each group put together a writeup on the following issues (probably ~1 page google doc).

- [x] Overview @rennergade 
- [x] glibc changes (MAKE_SYSCALL, crt generation, struct changes, etc) @yzhang71 @robinyuan1002 
- [x] Multi-processing/Multi-threading (pthreads, fork, exec, wait, longjmp) @qianxichen233 
- [ ] Memory management (vmmap and calls) @ChinmayShringi 
- [x] RawPOSIX (syscall api, cage structure, raw syscalls vs userspace syscalls, testing) @Yaxuan-w @ChinmayShringi 
- [ ] program compilation/application builds @robinyuan1002 @qianxichen233

**Comment by rennergade** [comment]
We have a short break coming up for Thanksgiving where people have a few days off or are traveling. If we can try to get these done around that time that would be awesome.

**Comment by rennergade** [comment]
> That would be great if we can have instructions on running the test / simple programs too

Yes strongly agree. Hopefully Robin and Yash can have that up and running soon with instructions.

**Comment by Yaxuan-w** [comment]
@ChinmayShringi  and I finished RawPOSIX part: https://docs.google.com/document/d/1D58-v092FJMcSmjg5dOyacE1iwq0SDGT1EquT0GmO64/edit?usp=sharing

**Comment by rennergade** [comment]
Added draft of overview section:
https://docs.google.com/document/d/1aMoNwBwjcEU9Ia2v3L5CefxnqHRNeDFtYKuqXkuN9fw/edit?tab=t.0

**Comment by yzhang71** [comment]
Here is the document for glibc part: https://docs.google.com/document/d/1aMoNwBwjcEU9Ia2v3L5CefxnqHRNeDFtYKuqXkuN9fw/edit?tab=t.0#heading=h.kx1q3fy634st

**Comment by rennergade** [comment]
@ruchjoshi-nyu Can you try to mock up something with the sections we've provided above?

**Comment by rennergade** [comment]
I reorganized the [docs repo](https://github.com/Lind-Project/lind-wasm-docs) yesterday. Thanks to @ruchjoshi-nyu  for helping organize some of this.

Some next steps

- It doesn't seem to be autobuilding due to actions being disabled possibly? @yashaswi2000 can you look into fixing this.
- the "libc" doc is pretty bare, and I left the old daily log and tls markdown documents in the repo even though we're no long serving them. @yzhang71 any chance you can try to summarize the important parts from those and add those to the libc.md file?
- I left a file for a bazel build doc pretty much empty @yashaswi2000 can you fill this out?
- There's a few markdowns now in the "use" folder for compilation, running, and debugging. I pretty much threw together multiple refs I had into each file so I'm not sure if everything is correct and/or if its redundant @qianxichen233 @robinyuan1002 can you guys look at those and update them?
- @ChinmayShringi can we add a doc in "internal" explaining how the memory management stuff works so far? Can just start with an overview of the vmmap

**Comment by rennergade** [comment]
@qianxichen233 can you add your pthread changes to the multi-processing doc, and also send @ChinmayShringi a few paragraphs for memory on the changes in your recent PR?

**Comment by rennergade** [comment]
I'm going to close this issue here and create individual issues in the docs repo. Will be easier to organize going forward.
